infrastructure/httpapi/handlers/crossfire.go

```diff
@@ -1,9 +1,162 @@
+	cosmosAppClient cosmosapp.Client,
+	rdbHandle *rdb.Handle,
+	participantsListURL string,
+) *Crossfire {
+	return &Crossfire{
+		logger.WithFields(applogger.LogFields{
+			"module": "CrossfireHTTPHander",
+		}),
+
+		validatorAddressPrefix,
+		consNodeAddressPrefix,
+
+		cosmosAppClient,
+		crossfire_validator_view.NewCrossfireValidators(rdbHandle),
+		crossfire_validator_view.NewCrossfireValidatorsStats(rdbHandle),
+		crossfire.NewHTTPClient(participantsListURL),
+	}
+}
+
+// ListAllCrossfireValidators Fetch all validators and combine them
+func (handler *Crossfire) ListAllCrossfireValidators(ctx *fasthttp.RequestCtx) {
+	var err error
+
+	// Fetch Remote List
+	participantList, errFetching := handler.httpClient.Participants()
+	if errFetching != nil {
+		handler.logger.Errorf("error listing validators: %v", errFetching)
+		httpapi.InternalServerError(ctx)
+	}
+
+	// Fetch DB Crossfire Validators List
+	crossfireValidatorsDB, err := handler.crossfireValidatorsView.List()
+	if err != nil {
+		handler.logger.Errorf("error listing validators: %v", err)
+		httpapi.InternalServerError(ctx)
+		return
+	}
+
+	var finalList = []CrossfireValidatorDetails{}
+
+	for _, participant := range *participantList {
+		for _, dbValidator := range crossfireValidatorsDB {
+
+			// For each matching participant create a return object
+			if dbValidator.OperatorAddress == participant.OperatorAddress && dbValidator.InitialDelegatorAddress == participant.PrimaryAddress {
+
+				validatorStats, err := handler.getValidatorStats(participant.OperatorAddress)
+				if err != nil {
+					continue
+				}
+				validatorDetail := CrossfireValidatorDetails{
+					operatorAddress:            participant.OperatorAddress,
+					initialDelegatorAddress:    dbValidator.InitialDelegatorAddress,
+					moniker:                    dbValidator.Moniker,
+					taskSetup:                  dbValidator.TaskPhase1NodeSetup,
+					taskKeepActive:             dbValidator.TaskPhase2KeepNodeActive,
+					taskProposalVote:           dbValidator.TaskPhase2ProposalVote,
+					taskNetworkUpgrade:         dbValidator.TaskPhase2NetworkUpgrade,
+					rankPhase1n2CommitmentRank: dbValidator.RankTaskPhase1n2CommitmentCount,
+					rankPhase3CommitmentRank:   dbValidator.RankTaskPhase3CommitmentCount,
+					rankTxSentRank:             dbValidator.RankTaskHighestTxSent,
+					stats:                      validatorStats,
+				}
+				finalList = append(finalList, validatorDetail)
+			}
+		}
+	}
+
+	httpapi.Success(ctx, finalList)
+}
+
+func (handler *Crossfire) getValidatorStats(operatorAddress string) (*ValidatorStats, error) {
+	// Fetch Validator Stats
+	key := fmt.Sprintf("%s%s", "%", operatorAddress)
+	validatorStatRows, err := handler.crossfireValidatorsStatsView.FindAllLike(key)
+	if err != nil {
+		return nil, fmt.Errorf("[error-crossfire] Fetching Validator Stats, %v", err)
+	}
+
+	var finalValidatorStats *ValidatorStats
+	for _, validatorStatRow := range validatorStatRows {
```

Comment:
Smart implementation! One case may need change is, the `PHASE1_COMMIT` is updated to `PHASE_1N2_COMMIT_PREFIX`

Reply:
Cool. Noted

Reply:
Btw, DOes that mean `PHASE_1N2_COMMIT_PREFIX ` it consists of both phase commit together ?

Reply:
yes, it consists of both phase commit together